feat: Add pt-br translation (#35)
open-sauced/intro: translations/pt-br/02-o-que-é-open-source.md

﻿# O que é Open Source?

Neste capítulo, vamos mergulhar no conceito de open source, sua definição e sua história. Obter uma compreensão profunda do open source ajudará você a apreciar o contexto em que estará a colaborar e contribuindo.

## Definição

O termo "open source" refere-se a um tipo de software cujo código-fonte é disponibilizado ao público, permitindo que qualquer pessoa visualize, use, modifique e distribua o código. Isso é em contraste com o software proprietário, onde o código-fonte é mantido em segredo e apenas o programa compilado é distribuído. O princípio básico por trás do código aberto é a ideia de colaboração e a crença de que o compartilhamento de conhecimento e recursos leva a um software melhor, mais confiável e mais inovador.

> Um software de código aberto é construído com base na premissa de que, quando muitas pessoas trabalham juntas, elas podem criar algo maior do que a soma das suas partes. Os contribuidores para projetos de código aberto vêm de origens diversas e trazem uma ampla gama de habilidades, experiências e perspetivas para a mesa. Essa diversidade de pensamento e experiência leva a um software mais robusto, inovador e confiável.

Reviewer: ```suggestion
Um software de código aberto é construído com base na premissa de que, quando muitas pessoas trabalham juntas, elas podem criar algo maior do que a soma das suas partes. Os contribuidores para projetos de código aberto vêm de origens diversas e trazem uma ampla gama de habilidades, experiências e perspectivas para a mesa. Essa diversidade de pensamento e experiência leva a um software mais robusto, inovador e confiável.
```
I guess both are acceptable but "perspectivas" is more common. Also, for consistency sake, let's use "perspectivas" everywhere

### Primeiros aparecimentos

Nos anos 50 e 60, o software de computador era frequentemente agrupado com hardware, e o código-fonte era compartilhado livremente entre pesquisadores e desenvolvedores. Durante esse período, era comum que os programadores colaborassem, compartilhassem ideias e modificassem o código uns dos outros para melhorar e inovar.

No entanto, na década de 1970, à medida que a indústria de software começou a crescer e a comercializar, as empresas começaram a ver o software como um ativo e começaram a restringir o acesso ao código-fonte. Essa mudança marcou o início da divisão entre software proprietário e o emergente movimento de código aberto.

### O nascimento do software livre

Como consequência da crescente restrição ao acesso ao código-fonte, o programador Richard Stallman fundou a Free Software Foundation (FSF) em 1985. O objetivo da FSF era promover o desenvolvimento e o uso de software livre, que eles definiram como software que concede aos usuários a liberdade de executar, estudar, compartilhar e modificar o código. Esse conceito é semelhante ao código aberto, mas enfatiza mais os aspectos éticos e políticos da liberdade de software.

> Stallman e a FSF passaram a desenvolver o Projeto GNU (que significa "GNU's Not Unix"), um esforço ambicioso para criar um sistema operativo completamente livre e aberto. Embora o Projeto GNU tenha fornecido muitos componentes essenciais, faltava uma peça crítica: um kernel livre e aberto, o componente central de um sistema operativo.

Reviewer: ```suggestion
Stallman e a FSF passaram a desenvolver o Projeto GNU (que significa "GNU's Not Unix"), um esforço ambicioso para criar um sistema operacional completamente livre e aberto. Embora o Projeto GNU tenha fornecido muitos componentes essenciais, faltava uma peça crítica: um kernel livre e aberto, o componente central de um sistema operacional.
```

À medida que o movimento de código aberto ganhava força, inúmeras comunidades e plataformas surgiram para facilitar a colaboração e o compartilhamento de conhecimento. Alguns exemplos notáveis incluem:

- **SourceForge**: Lançado em 1999, o SourceForge foi uma das primeiras plataformas baseadas na web a fornecer ferramentas e recursos para que os desenvolvedores colaborassem em projetos de código aberto. No auge, o SourceForge hospedava milhares de projetos e atraía milhões de usuários.
- **GitHub**: Fundado em 2008, o GitHub rapidamente se tornou a plataforma padrão para hospedar e colaborar em repositórios Git. Hoje, o GitHub é o lar de milhões de projetos de código aberto e possui uma comunidade vibrante de colaboradores de todo o mundo.
- **GitLab**: Lançado em 2011 como uma alternativa de código aberto ao GitHub, o GitLab oferece um conjunto de ferramentas e recursos para gerir repositórios Git, rastrear problemas e facilitar a colaboração.

Essas plataformas desempenharam um papel crucial no fomento ao crescimento das comunidades de código aberto e na facilitação da contribuição e colaboração de desenvolvedores em projetos.

### O crescimento do envolvimento corporativo

> Nos últimos anos, houve um aumento significativo do envolvimento corporativo em projetos de código aberto. Muitas empresas agora reconhecem o valor do código aberto e contribuem ativamente e apoiam vários projetos. Algumas empresas até foram tão longe a ponto de abrir seus próprios ferramentas e tecnologias internas, como o TensorFlow do Google e o React do Facebook.

Reviewer: ```suggestion
Nos últimos anos, houve um aumento significativo do envolvimento corporativo em projetos de código aberto. Muitas empresas agora reconhecem o valor do código aberto e contribuem ativamente e apoiam vários projetos. Algumas empresas até foram tão longe a ponto de abrir suas próprias ferramentas e tecnologias internas, como o TensorFlow do Google e o React do Facebook.
```

Este aumento do envolvimento corporativo não só levou a mais recursos e suporte para projetos de código aberto, mas também ajudou a legitimar o movimento de código aberto e incentivar uma adoção mais ampla.

### A evolução do código aberto

O movimento de código aberto percorreu um longo caminho desde os seus primórdios. Hoje, o software de código aberto está em toda parte, alimentando desde sites e aplicativos móveis até algoritmos de inteligência artificial e aprendizado de máquina.

À medida que o código aberto continua a evoluir, novos projetos, ferramentas e tecnologias estão constantemente surgindo, empurrando os limites do que é possível e impulsionando a inovação em uma ampla gama de indústrias.

> O crescente interesse e adoção do código aberto também levaram a ecossistema de código aberto próspero de conferências, meetups e comunidades online onde os desenvolvedores podem se reunir para aprender, colaborar e compartilhar os seus conhecimentos. Ao participar desses eventos e se envolver com a comunidade de código aberto, você terá a oportunidade de expandir as suas habilidades, construir a sua rede e se manter atualizado com os últimos desenvolvimentos na �rea.

Reviewer: ```suggestion
O crescente interesse e adoção do código aberto também levaram a um ecossistema de código aberto próspero de conferências, meetups e comunidades online onde os desenvolvedores podem se reunir para aprender, colaborar e compartilhar os seus conhecimentos. Ao participar desses eventos e se envolver com a comunidade de código aberto, você terá a oportunidade de expandir as suas habilidades, construir a sua rede e se manter atualizado com os últimos desenvolvimentos na área.
```
